scripts/run_action.mjs

```diff
@@ -14,87 +14,169 @@
-      buffer
-        .toString()
-        .split('\n')
-        .filter(line => line.trim())
-        .forEach(callback);
-    };
-
-    childProcess.stdout.on('data', data => forEachMessageLine(data, line => console.info(line)));
-    childProcess.stderr.on('data', error => forEachMessageLine(error, line => console.error(chalk.red(line))));
-
-    childProcess.on('close', code => {
-      if (code === 0) {
-        resolve(childProcess);
-      } else {
-        reject(childProcess);
-      }
-    });
-  });
+  async update(actionPathKey, actionCacheObject) {
+    let cache = {};
+
+    if (existsSync(this.cachePath)) {
+      cache = JSON.parse(await fs.readFile(this.cachePath));
+    }
+
+    cache[actionPathKey] = actionCacheObject;
+
+    return fs.writeFile(this.cachePath, JSON.stringify(cache));
+  }
+}
 
 /**
  * @description This is the entrypoint into our custom task runner.
  * It runs an `*.action.mjs` or `*.action.sh` file with the given parameters.
  * You can find these files located within their associated directories in the project.
- * Action files can also define:
- *   `requirements` - actions that must be run beforehand as prerequisite.
- *   `dependencies` - files and folders that the action takes as input. If they remain unchanged, the action can be skipped.
  *
  * @param {string} actionPath The truncated path to the action you wish to run (e.g. "www/build")
- * @param {...string} parameters The flags and other parameters we want to run the action with.
+ * @param {Object} options Additional action options
+ * @param {string[]} options.parameters Command line arguments passed in to the action script
+ * @param {string[]} options.inputs Files and folders the action consumes. Specify these to skip the action when able
  * @returns {void}
  */
-export async function runAction(actionPath, ...parameters) {
-  const resolvedPath = await resolveActionPath(actionPath);
+export async function runAction(actionPath, {parameters = [], inputs = []} = {}) {
+  /**
+   * @description promisifies the child process spawner
+   */
+  const spawnPromise = (command, parameters) =>
+    new Promise((resolve, reject) => {
+      const childProcess = spawn(command, parameters, {shell: true});
+
+      const forEachMessageLine = (buffer, callback) => {
+        buffer
+          .toString()
+          .split('\n')
+          .filter(line => line.trim())
+          .forEach(callback);
+      };
+
+      childProcess.stdout.on('data', data => forEachMessageLine(data, line => console.info(line)));
+      childProcess.stderr.on('data', error => forEachMessageLine(error, line => console.error(chalk.red(line))));
+
+      childProcess.on('close', code => {
+        if (code === 0) {
+          resolve(childProcess);
+        } else {
+          reject(childProcess);
+        }
+      });
+    });
+
+  let resolvedPath;
+  const pathRoots = [process.env.ROOT_DIR, path.join(process.env.ROOT_DIR, 'src')];
```

Comment:
Can you move these to some `resolveActionPath` function?
It will help the readability of `runAction`